apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClientImpl.cs

```diff
@@ -0,0 +1,702 @@
+        private static void ValidateCustomMetadataKey(string key)
+        {
+            if (!KeyRegex.IsMatch(key))
+            {
+                throw new ArgumentException("Invalid custom metadata key.", "key");
+            }
+        }
+
+        /// <summary>
+        /// Gets a token response which contains an access token to authorize a request synchronously.
+        /// </summary>
+        /// <exception cref="TokenResponseException">A status code other than success was returned from the server.</exception>
+        /// <returns>The token response.</returns>
+        public override TokenResponse GetAccessToken() =>
+            GetResult(() => GetAccessTokenAsync());
+
+        /// <summary>
+        /// Gets a token response which contains an access token to authorize a request asynchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <exception cref="TokenResponseException">A status code other than success was returned from the server.</exception>
+        /// <returns>The token response.</returns>
+        public override async Task<TokenResponse> GetAccessTokenAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // TODO: Most of this is taken from Google.Apis.Auth.OAuth2.ComputeCredential.RequestAccessTokenAsync. Perhaps move that to a helper which can be shared.
+            var response = await RequestMetadata("instance/service-accounts/default/token", cancellationToken).ConfigureAwait(false);
+
+            // Read the response.
+            var content = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+            if (!response.IsSuccessStatusCode)
+            {
+                try
+                {
+                    var error = NewtonsoftJsonSerializer.Instance.Deserialize<TokenErrorResponse>(content);
+                    throw new TokenResponseException(error);
+                }
+                // Json.NET deserialize object method doesn't include documentation about which exceptions can
+                // be thrown, that's why we just catch a general exception here.
+                catch (Exception ex)
+                {
+                    Logger.Error(ex, "Exception was caught when deserializing TokenErrorResponse. Content is: {0}",
+                        content);
+                    var error = "Server response does not contain a JSON object. Status code is: "
+                        + response.StatusCode;
+                    throw new TokenResponseException(new TokenErrorResponse
+                    {
+                        Error = error
+                    });
+                }
+            }
+
+            // Gets the token and sets its issued time.
+            var newToken = NewtonsoftJsonSerializer.Instance.Deserialize<TokenResponse>(content);
+            newToken.Issued = DateTime.Now; // TODO: This was using an IClock before.
```

Comment:
This should use DateTime.UtcNow, not DateTime.Now. (Find any other occurrences, too...)